payment/src/main/java/com/kyriba/conference/payment/service/PaymentMethodServiceImpl.java

```diff
@@ -0,0 +1,54 @@
+package com.kyriba.conference.payment.service;
+
+import com.kyriba.conference.payment.api.dto.PaymentMethodDto;
+import com.kyriba.conference.payment.api.dto.PaymentMethodResponse;
+import com.kyriba.conference.payment.api.dto.PaymentMethodUrlUpdateDto;
+import com.kyriba.conference.payment.dao.PaymentMethodRepository;
+import com.kyriba.conference.payment.domain.PaymentMethodEntity;
+import com.kyriba.conference.payment.domain.PaymentMethodType;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+@AllArgsConstructor
+@Transactional
+public class PaymentMethodServiceImpl implements PaymentMethodService {
+    private final PaymentMethodRepository paymentMethodRepository;
+
+    @Override
+    public List<PaymentMethodDto> findAll() {
+        return paymentMethodRepository.findAll().stream()
+                .map(PaymentMethodEntity::toPaymentMethodDto).collect(Collectors.toList());
+    }
+
+    @Override
+    public PaymentMethodResponse createPaymentMethod(PaymentMethodDto paymentMethodDto) {
```

Comment:
@xpinjection should we use Transactional(readOnly=true) for 'read' methods?

Reply:
This is performance tuning, so optional at that stage.